### PR TITLE
Enable router ip update

### PR DESCRIPTION
**Pull request by mjudeikis**
### Which issue this PR addresses:

Fixes cases where customer recreated `router-default` and external IP changes

### What this PR does / why we need it:

This gives us tools to update this without JIT

### Test plan for issue:

No

### Is there any documentation that needs to be updated for this PR?

Yes, SOP

### CODE REVIEW

File: pkg/cluster/ipaddresses.go [hunk 1]
```diff
@@ -25,6 +25,16 @@ func (m *manager) updateRouterIP(ctx context.Context) error {
...
+	m.doc, err = m.db.PatchWithLease(ctx, m.doc.Key, func(doc *api.OpenShiftClusterDocument) error {
+		doc.OpenShiftCluster.Properties.APIServerProfile.URL = "https://api." + installConfig.Config.ObjectMeta.Name + "." + installConfig.Config.BaseDomain + ":6443/"
+		doc.OpenShiftCluster.Properties.ConsoleProfile.URL = "https://console-openshift-console.apps." + installConfig.Config.ObjectMeta.Name + "." + installConfig.Config.BaseDomain + "/"
+		doc.OpenShiftCluster.Properties.KubeadminPassword = api.SecureString(kubeadminPassword.Password)
+		return nil
+	})
+	return err
+}
+
+func (m *manager) updateRouterIP(ctx context.Context) error {
```
Comment (jim-minter) [inline pkg/cluster/ipaddresses.go:37]: 1. the following code is appropriate in the context of install, but is it appropriate that it should cause an error in adminupgrade?

```golang
	if len(svc.Status.LoadBalancer.Ingress) == 0 {
		return fmt.Errorf("routerIP not found")
	}
```

2. In admin upgrade I would prefer to see us test the router IP address to see if it needs changing, and only change it if it really does.  It saves a gratuitous API write call per cluster.

File: pkg/cluster/ipaddresses.go [hunk 2]
```diff
@@ -16,7 +16,7 @@ import (
 	"github.com/Azure/ARO-RP/pkg/util/stringutils"
 )
 
-func (m *manager) updateRouterIP(ctx context.Context) error {
+func (m *manager) updateClusterData(ctx context.Context) error {
```
Comment (jim-minter) [inline pkg/cluster/ipaddresses.go:19]: I am almost tempted to roll this code into the PatchWithLease at the bottom of deploystorage.go.

File: pkg/cluster/ipaddresses.go (hunk 1 above)
Comment (jim-minter) [reply pkg/cluster/ipaddresses.go:37]: 3. Also, what happens if there's no router-default service?

Comment (jim-minter) [reply pkg/cluster/ipaddresses.go:37]: > test the router IP address to see if it needs changing
Amber suggests that code might want to live in the DNS manager.

File: pkg/cluster/ipaddresses.go (hunk 2 above)
Comment (mjudeikis) [reply pkg/cluster/ipaddresses.go:19]: What is the concert here? Too many calls? 
I think this segregates details nicely if case we need to drop actions (not like we gonna do this any time soon)

**Pull request merged**